docusaurus/docs/create-a-plugin/extend-a-plugin/instrument-a-backend-plugin.md

```diff
@@ -0,0 +1,421 @@
+                   attribute.String("my_plugin.my_attribute", "custom value"),
+               },
+           },
+       }); err != nil {
+           log.DefaultLogger.Error(err.Error())
+           os.Exit(1)
+       }
+   }
+   ```
+
+1. Once you have configured tracing, use the global tracer like this:
+
+   ```go
+   tracing.DefaultTracer()
+   ```
+
+   This returns an [OpenTelemetry `trace.Tracer`](https://pkg.go.dev/go.opentelemetry.io/otel/trace#Tracer) for creating spans.
+
+   **Example:**
+
+   ```go
+   func (d *Datasource) query(ctx context.Context, pCtx backend.PluginContext, query backend.DataQuery) (backend.DataResponse, error) {
+       ctx, span := tracing.DefaultTracer().Start(
+           ctx,
+           "query processing",
+           trace.WithAttributes(
+               attribute.String("query.ref_id", query.RefID),
+               attribute.String("query.type", query.QueryType),
+               attribute.Int64("query.max_data_points", query.MaxDataPoints),
+               attribute.Int64("query.interval_ms", query.Interval.Milliseconds()),
+               attribute.Int64("query.time_range.from", query.TimeRange.From.Unix()),
+               attribute.Int64("query.time_range.to", query.TimeRange.To.Unix()),
+           ),
+       )
+       defer span.End()
+
+       // ...
+   }
+   ```
+
+### Automatic instrumentation by the SDK
+
+The SDK provides some automatic instrumentation to ease developer experience.
+
+#### Tracing gRPC calls
+
+When tracing is enabled, a new span is created automatically for each gRPC call (`QueryData`, `CallResource`, `CheckHealth`, etc.), both on Grafana's side and on the plugin's side. The plugin SDK also injects the trace context into the `context.Context` that is passed to those methods.
+
+You can retrieve the [trace.SpanContext](https://pkg.go.dev/go.opentelemetry.io/otel/trace#SpanContext) with `tracing.SpanContextFromContext` by passing the original `context.Context` to it:
+
+```go
+func (d *Datasource) query(ctx context.Context, pCtx backend.PluginContext, query backend.DataQuery) (backend.DataResponse, error) {
+    spanCtx := trace.SpanContextFromContext(ctx)
+    traceID := spanCtx.TraceID()
+
+    // ...
+}
+```
+
+#### Tracing outgoing HTTP requests
+
+When tracing is enabled, a `TracingMiddleware` is also added to the default middleware stack to all HTTP clients created using the [httpclient.New](https://pkg.go.dev/github.com/grafana/grafana-plugin-sdk-go/backend/httpclient#New) or [httpclient.NewProvider](https://pkg.go.dev/github.com/grafana/grafana-plugin-sdk-go/backend/httpclient#NewProvider), unless you specify custom middleware. This middleware creates spans for each outgoing HTTP request and provides some useful attributes and events related to the request's lifecycle.
+
+### Collect and visualize traces locally
+
+Please refer to [How to collect and visualize logs, metrics and traces](#collect-and-visualize-logs-metrics-and-traces).
```

Comment:
```suggestion
Refer to [How to collect and visualize logs, metrics and traces](#collect-and-visualize-logs-metrics-and-traces).
```

Reply:
Generally, use "please" only when Grafana has inconvenienced a user and so they must do something additionally.